src/Fixer/Comment/NoEmptyCommentFixer.php

```diff
@@ -58,34 +67,105 @@ public function getPriority()
-        if ('#' === $content[0]) {
-            if (preg_match('|^#\s*$|', $content)) {
-                $tokens->clearTokenAndMergeSurroundingWhitespace($index);
+        for (; $index < $count; ++$index) {
+            if ($tokens[$index]->isComment()) {
+                if ($commentType !== $this->getCommentType($tokens[$index]->getContent())) {
+                    break;
+                }
+
+                if ($empty) { // don't retest if already known the block not being empty
+                    $empty = $this->isEmptyComment($tokens[$index]->getContent());
+                }
+
+                continue;
             }
 
-            return;
+            if (!$tokens[$index]->isWhitespace() || $this->getLineBreakCount($tokens, $index, $index + 1) > 1) {
+                break;
+            }
         }
 
-        // single line comment starting with '//'
-        if ('/' === $content[1]) {
-            if (preg_match('|^//\s*$|', $content)) {
-                $tokens->clearTokenAndMergeSurroundingWhitespace($index);
-            }
+        return array($start, $index - 1, $empty);
+    }
+
+    /**
+     * @param string $content
+     *
+     * @return int
+     */
+    private function getCommentType($content)
+    {
+        if ('#' === $content[0]) {
+            return self::TYPE_HASH;
+        }
 
-            return;
+        if ('*' === $content[1]) {
+            return self::TYPE_SLASH_ASTERISK;
         }
 
-        // comment starting with '/*' and ending with '*/' (but not a PHPDoc)
-        if (preg_match('|^/\*\s*\*/$|', $content)) {
-            $tokens->clearTokenAndMergeSurroundingWhitespace($index);
+        return self::TYPE_DOUBLE_SLASH;
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $whiteStart
+     * @param int    $whiteEnd
+     *
+     * @return int
+     */
+    private function getLineBreakCount(Tokens $tokens, $whiteStart, $whiteEnd)
+    {
+        $lineCount = 0;
+        for ($i = $whiteStart; $i < $whiteEnd; ++$i) {
+            $lineCount += substr_count($tokens[$i]->getContent(), "\n");
+        }
+
+        return $lineCount;
+    }
+
+    /**
+     * @param string $content
+     *
+     * @return bool
+     */
+    private function isEmptyComment($content)
+    {
+        $type = $this->getCommentType($content);
+        switch ($type) {
+            case self::TYPE_HASH:
+                // single line comment starting with '#'
+                return 1 === preg_match('|^#\s*$|', $content);
+            case self::TYPE_SLASH_ASTERISK:
+                // comment starting with '/*' and ending with '*/' (but not a PHPDoc)
+                return 1 === preg_match('|^/\*\s*\*/$|', $content);
+            case self::TYPE_DOUBLE_SLASH:
+                // single line comment starting with '//'
+                return 1 === preg_match('|^//\s*$|', $content);
+            default:
+                throw new \UnexpectedValueException('Comment type detected failed.');
```

Comment:
could never happen, as getCommentType return only 3 values (if it cannot detect it returns `//`)